quicvarint/varint.go

```diff
@@ -28,53 +26,53 @@ func Read(b io.ByteReader) (uint64, error) {
 	if len == 1 {
 		return uint64(b1), nil
 	}
-	b2, err := b.ReadByte()
+	b2, err := r.ReadByte()
 	if err != nil {
 		return 0, err
 	}
 	if len == 2 {
 		return uint64(b2) + uint64(b1)<<8, nil
 	}
-	b3, err := b.ReadByte()
+	b3, err := r.ReadByte()
 	if err != nil {
 		return 0, err
 	}
-	b4, err := b.ReadByte()
+	b4, err := r.ReadByte()
 	if err != nil {
 		return 0, err
 	}
 	if len == 4 {
 		return uint64(b4) + uint64(b3)<<8 + uint64(b2)<<16 + uint64(b1)<<24, nil
 	}
-	b5, err := b.ReadByte()
+	b5, err := r.ReadByte()
 	if err != nil {
 		return 0, err
 	}
-	b6, err := b.ReadByte()
+	b6, err := r.ReadByte()
 	if err != nil {
 		return 0, err
 	}
-	b7, err := b.ReadByte()
+	b7, err := r.ReadByte()
 	if err != nil {
 		return 0, err
 	}
-	b8, err := b.ReadByte()
+	b8, err := r.ReadByte()
 	if err != nil {
 		return 0, err
 	}
 	return uint64(b8) + uint64(b7)<<8 + uint64(b6)<<16 + uint64(b5)<<24 + uint64(b4)<<32 + uint64(b3)<<40 + uint64(b2)<<48 + uint64(b1)<<56, nil
 }
 
-// Write writes a number in the QUIC varint format
-func Write(b *bytes.Buffer, i uint64) {
+// Write writes i in the QUIC varint format to w.
+func Write(w Writer, i uint64) {
```

Comment:
Same here.

Reply:
`Write` uses both `Write` and `WriteByte`.

Reply:
True, sorry, I missed that.